.github/workflows/unit-tests.yml

```diff
@@ -22,39 +23,76 @@ jobs:
+          #   GCC 9 also known to show a few warnings that newer versions dont show
+          #   Use (compiler) default C++ standard (14)
+          #   Use system cmake
+          #   Use system boost (min version)
+          - { compiler: gcc-9,    os: ubuntu-20.04, type: Debug, cmake: 3.16.3, boost: 1.69.0 }
+          - { compiler: clang-10, os: ubuntu-20.04, type: Debug, cmake: 3.16.3, boost: 1.69.0, externalSanitizer: true }
+          #
+          #   Default compiler for Ubuntu 20.04
+          #   Use (compiler) default C++ standard (14)
+          #   Use default cmake
+          #   Use default boost
+          - { compiler: gcc-10,   os: ubuntu-20.04, type: Debug, coverage: true }
+          #
+          #   Default compilers for Ubuntu 22.04
+          #   Use C++ standard (17, default for gcc)
+          #   Use system cmake
+          #   Use system boost
+          - { compiler: gcc-11,   os: ubuntu-22.04, type: Debug, cmake: 3.22.1, boost: 1.74.0 }
+          - { compiler: clang-14, os: ubuntu-22.04, type: Debug, cmake: 3.22.1, boost: 1.74.0, cxx: 17 }
+          #
+          #  Latest Compilers
+          #   GCC 12 needs boost 1.82 to compile correctly
+          #   Use (compiler) default C++ standard (17)
+          #   Use default cmake
+          - { compiler: gcc-12,   os: ubuntu-22.04, type: Debug, boost: 1.82.0 }
+          #   Use default boost
+          - { compiler: clang-16, os: ubuntu-20.04, type: Debug, externalSanitizer: true }
 
     runs-on: ${{matrix.os}}
 
     steps:
       - run: echo "DEPS_DIR=${{runner.temp}}/.cache" >> $GITHUB_ENV
-      - run: echo "CMAKE_VERSION=${{matrix.cmakeVersion}}" >> $GITHUB_ENV
-        if: matrix.cmakeVersion
-      - run: echo "BOOST_VERSION=${{matrix.boostVersion}}" >> $GITHUB_ENV
-        if: matrix.boostVersion
-      - run: echo "ADDITIONAL_CMAKE_FLAGS=${ADDITIONAL_CMAKE_FLAGS} -DRTTR_EXTERNAL_BUILD_TESTING=ON -DRTTR_ENABLE_SANITIZERS=ON" >> $GITHUB_ENV
+
+      - name: "Set BUILD_TYPE"
+        run: echo "BUILD_TYPE=${{matrix.type}}" >> $GITHUB_ENV
+
+      - name: "Set CMake version"
+        if: matrix.cmake
+        run: echo "CMAKE_VERSION=${{matrix.cmake}}" >> $GITHUB_ENV
+
+      - name: "Set Boost version"
+        if: matrix.boost
+        run: echo "BOOST_VERSION=${{matrix.boost}}" >> $GITHUB_ENV
+
+      - name: Set C++ standard
+        if: matrix.cxx
+        run: echo "ADDITIONAL_CMAKE_FLAGS=${ADDITIONAL_CMAKE_FLAGS} -DCMAKE_CXX_STANDARD=${{matrix.cxx}}" >> $GITHUB_ENV
+
+      - name: Enable coverage collection
+        if: matrix.coverage
+        run: echo "ADDITIONAL_CMAKE_FLAGS=${ADDITIONAL_CMAKE_FLAGS} -DRTTR_ENABLE_COVERAGE=ON" >> $GITHUB_ENV
+
+      - name: "Enable external sanitizer"
         if: matrix.externalSanitizer
+        run: echo "ADDITIONAL_CMAKE_FLAGS=${ADDITIONAL_CMAKE_FLAGS} -DRTTR_EXTERNAL_BUILD_TESTING=ON -DRTTR_ENABLE_SANITIZERS=ON" >> $GITHUB_ENV
 
+      # Coverage collection requires access to history to find merge-base
```

Comment:
Why move this above the part that does **not** apply to coverage? Then I'd swap the 2 so this looks more logical and maybe add a comment like  `fetch-depth: 0  # Faster` or so while at it

Reply:
so that the comment is above the block